### PR TITLE
added: view text files in file manager

### PR DESCRIPTION
**Pull request by notspiff**
as default selection action for .txt and .xml files
for everything else add a context menu entry

### CODE REVIEW

**Comment by notspiff** [comment]
ehm, just realized this depends on #13092 due to the monospace requesting.

**Review by xhaggi — APPROVED** [review]

**Comment by garbear** [comment]
what happens if you try to view a binary file?

File: xbmc/windows/GUIWindowFileManager.cpp [hunk 1]
```diff
@@ -1279,3 +1289,24 @@ const CFileItem& CGUIWindowFileManager::CurrentDirectory(int indx) const
...
+void CGUIWindowFileManager::TextView(const std::string& path)
+{
+  CFile file;
+  if (file.Open(path))
+  {
+    std::string data;
+    data.resize(file.GetLength()+1);
+    if (data.size() <= static_cast<size_t>(file.GetLength()))
+    {
+      CLog::Log(LOGERROR, "Not enough memory to load text file %s", path.c_str());
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/windows/GUIWindowFileManager.cpp]: yes it ofc it will... fixed.

**Comment by notspiff** [comment]
the same thing that happens if you try to open it in some random text editor.

File: xbmc/windows/GUIWindowFileManager.cpp [hunk 2]
```diff
@@ -1279,3 +1289,30 @@ const CFileItem& CGUIWindowFileManager::CurrentDirectory(int indx) const
 {
   return *m_Directory[indx];
 }
+
+void CGUIWindowFileManager::TextView(const std::string& path)
```
Comment (phil65) [inline xbmc/windows/GUIWindowFileManager.cpp]: since this method is static, perhaps we could move it to some utility class?

**Review by phil65 — APPROVED** [review]

File: xbmc/windows/GUIWindowFileManager.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/windows/GUIWindowFileManager.cpp]: moved to a static.

**Comment by notspiff** [comment]
jenkins build this please

**Pull request merged**